src/base/bittorrent/tracker.cpp

```diff
@@ -140,8 +140,11 @@ void Tracker::respondToAnnounceRequest()
         const int sepPos = param.indexOf('=');
         if (sepPos <= 0) continue; // ignores params without name
 
-        const QString paramName {QString::fromUtf8(param.constData(), sepPos)};
-        const QByteArray paramValue {param.mid(sepPos + 1)};
+        const QByteArray nameComponent = midView(param, 0, sepPos);
+        const QByteArray valueComponent = midView(param, (sepPos + 1));
+
+        const QString paramName = QString::fromUtf8(QByteArray::fromPercentEncoding(nameComponent));
```

Comment:
It seems to me that it will look a bit confusing... When looking at this code, I have a question, how does it work? Won't it return an empty string because we're using it with an invalid URL?

Reply:
I read the docs most of the time when calling Qt function, so I'm fine with both styles, up to you.

Reply:
>I read the docs most of the time when calling Qt function

Me neither. But nothing prevents Qt developers from changing its behavior to the one I described above, since it is a member of the QUrl class. So I prefer to leave it in this explicit way.